Ligandr/LigOverlay.py

```diff
@@ -0,0 +1,217 @@
+            new_structure.SetTitle("ligand replacement")
+            docked_ligand = __docking_internal(receptor, ligand_posed, new_structure)
+            docked_ligand.SetTitle(title_list[i][j])
+            ligand_position.append(docked_ligand)
+            j = j + 1
+        ligand_stream.append(ligand_position)
+        overall_dry_frame.atom_slice(overall_dry_frame.top.select("not ( %s )" % ligand), inplace=True)
+        i = i + 1
+    # Produce output trajectories for all combos of ligands
+    for ligand_combo in itertools.product(*ligand_stream):
+        new_frame_prot = overall_dry_frame  # type: Trajectory
+        for built_ligand in ligand_combo:
+            # Rename ligand
+            ligand_dummy = tempfile.NamedTemporaryFile(suffix=".pdb")
+            outputter_lig = oechem.oemolostream(ligand_dummy.name)
+            oechem.OEWriteMolecule(outputter_lig, built_ligand)
+            ligand_final_traj = mdtraj.load(ligand_dummy.name)
+            ligand_dummy.close()
+            for residue in ligand_final_traj.top.residues:
+                residue.name = built_ligand.GetTitle()
+            new_frame_prot = new_frame_prot.stack(ligand_final_traj)
+        output.append(new_frame_prot)
+    print("End")
+    return output
+
+
+def __docking_internal(receptor: oechem.OEGraphMol, bound_ligand: oechem.OEGraphMol, docking_ligand: oechem.OEGraphMol):
+    """
+    Internal method for docking to a receptor given a bound ligand and ligand to dock to
+
+    Parameters
+    ----------
+    receptor: oechem.OEGraphMol, required
+        An oechem receptor to dock to
+
+    bound_ligand: oechem.OEGraphMol, required
+        The ligand bound to the current receptor pocket
+
+    docking_ligand: oechem.OEGraphMol, required
+        The ligand to dock.
+
+    Returns
+    -------
+    A oechem molecules with coordinates of its docking
+
+    """
+    # Create posit config
+    oedocking.OEReceptorSetBoundLigand(receptor, bound_ligand)
+    poser = oedocking.OEPosit()
+    poser.Initialize(receptor)
+
+    # Create multiple conformations
+    omegaOpts = oeomega.OEOmegaOptions()
+    omegaOpts.SetMaxConfs(1000)
+    omega_driver = oeomega.OEOmega(omegaOpts)
+    conformer_docking = oechem.OEMol()  # type: OEMol
+    if not omega_driver(conformer_docking):
+        single_sdf = tempfile.NamedTemporaryFile(suffix=".sdf")
+        double_sdf = tempfile.NamedTemporaryFile(suffix=".sdf")
+        smiles = NameToSMILES.addH(oechem.OEMolToSmiles(docking_ligand))
+        subprocess.run("obabel -:'%s' -O %s --gen3d; obabel %s -O %s --confab --conf = 100" % (
+            smiles, single_sdf.name, single_sdf.name, double_sdf.name), shell=True)
+        ins = oechem.oemolistream()
+        ins.SetConfTest(oechem.OEOmegaConfTest())
+        ins.open(double_sdf.name)
+        oechem.OEReadMolecule(ins, conformer_docking)
+
+    # Failed conformer generation
+    if conformer_docking.NumConfs() <= 1:
```

Comment:
Unsure if this test should be for `<= 1` or `< 1` -- confirm?

Reply:
It should be less than equal to one because openeye still retains a blank conformer if generation fails.

Reply:
Ahh, thanks for confirming!